src/components/BankTransactions/BankTransactionsHeader.tsx

```diff
@@ -148,30 +153,96 @@ export const BankTransactionsHeader = ({
   const { count } = useCountSelectedIds()
   const { clearSelection } = useBulkSelectionActions()
 
+  const [isConfirmAllModalOpen, setIsConfirmAllModalOpen] = useState(false)
+  const [isCategorizeAllModalOpen, setIsCategorizeAllModalOpen] = useState(false)
+  const [selectedCategory, setSelectedCategory] = useState<CategoryOption | undefined>(undefined)
+
+  const categorySelectId = useId()
+
+  const handleConfirmAllClick = useCallback(() => {
+    setIsConfirmAllModalOpen(true)
+  }, [])
+
+  const handleCategorizeAllClick = useCallback(() => {
+    setIsCategorizeAllModalOpen(true)
+  }, [])
+
+  const handleCategorizeModalClose = useCallback((isOpen: boolean) => {
+    setIsCategorizeAllModalOpen(isOpen)
+    if (!isOpen) {
+      setSelectedCategory(undefined)
+    }
+  }, [])
+
   const BulkActions = useCallback(() => {
     return (
       <HStack align='center' gap='sm'>
         <Button
           variant='outlined'
-          // TODO: onClick
+          onClick={handleConfirmAllClick}
         >
-          Categorize
+          Confirm All
         </Button>
+        <BaseConfirmationModal
+          isOpen={isConfirmAllModalOpen}
+          onOpenChange={setIsConfirmAllModalOpen}
+          title='Confirm all selected suggestions?'
```

Comment:
Let's do a content design pass here with Justin. Maybe something like
```suggestion
          title='Confirm all suggestions?'
```